homeassistant/components/netatmo/climate.py

```diff
@@ -82,90 +143,274 @@ def temperature_unit(self):
+    def __init__(self, auth, home=None):
         """Initialize the data object."""
         self.auth = auth
-        self.thermostatdata = None
-        self.module_names = []
-        self.device = device
-        self.current_temperature = None
-        self.target_temperature = None
-        self.setpoint_mode = None
-
-    def get_module_names(self):
+        self.homedata = None
+        self.homestatus = None
+        self.room_ids = []
+        self.room_status = {}
+        self.schedules = []
+        self.home = home
+        self.away_temperature = None
+        self.hg_temperature = None
+        self.boilerstatus = None
+        self.setpoint_duration = None
+        self.home_id = None
+
+    def get_room_ids(self):
         """Return all module available on the API as a list."""
-        self.update()
-        if not self.device:
-            for device in self.thermostatdata.modules:
-                for module in self.thermostatdata.modules[device].values():
-                    self.module_names.append(module['module_name'])
-        else:
-            for module in self.thermostatdata.modules[self.device].values():
-                self.module_names.append(module['module_name'])
-        return self.module_names
+        if self.setup():
+            for key in self.homestatus.rooms:
+                self.room_ids.append(key)
+            return self.room_ids
+        return []
+
+    def setup(self):
+        """Retrieve HomeData and HomeStatus by NetAtmo API."""
+        import pyatmo
+        try:
+            self.homedata = pyatmo.HomeData(self.auth)
+            self.homestatus = pyatmo.HomeStatus(self.auth, home=self.home)
+            self.home_id = self.homedata.gethomeId(self.home)
+            self.update()
+        except TypeError:
+            _LOGGER.error("ThermostatData::setup() got error.")
+            return False
+        return True
 
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
     def update(self):
         """Call the NetAtmo API to update the data."""
         import pyatmo
-        self.thermostatdata = pyatmo.ThermostatData(self.auth)
-        self.target_temperature = self.thermostatdata.setpoint_temp
-        self.setpoint_mode = self.thermostatdata.setpoint_mode
-        self.current_temperature = self.thermostatdata.temp
+        try:
+            self.homestatus = pyatmo.HomeStatus(self.auth, home=self.home)
+        except TypeError:
+            _LOGGER.error("Error when getting homestatus.")
+            return
+        _LOGGER.debug("Following is the debugging output for homestatus:")
+        _LOGGER.debug(self.homestatus.rawData)
+        for key in self.homestatus.rooms:
+            roomstatus = {}
+            roomstatus['roomID'] = self.homestatus.rooms[key]['id']
+            roomstatus['roomname'] = \
+                self.homedata.rooms[self.home][key]['name']
+            roomstatus['target_temperature'] = \
+                self.homestatus.rooms[key]['therm_setpoint_temperature']
```

Comment:
Pull `self.homestatus.rooms[key]` to it's own variable since it's referenced commonly.

Pull `self.homedata.rooms[self.home][key]` to it's own variable since it's reference commonly.

Reply:
Resolved in 4515f57.